server/app/api/user/user_service.py

```diff
@@ -82,7 +86,14 @@ def exists_with_credentials(self, username: str, password: str) -> bool:
         if not verify_password(password, user.hashed_password):
             raise False
 
-        return True  
+        return True
+
+    def search_by_username_prefix(self, query: str, org_id: int) -> List[User]:
+        all_users = self.user_repo.search_by_username_prefix(query)
+        users_to_exclude = self.org_repo.find_members_of_org(org_id)
+        users = [u for u in all_users if u not in users_to_exclude]
+        return users
+
```

Comment:
`org_id` should be renamed to better explain what the field does, like `org_id_whose_members_to_exclude` (just like in `user.api.ts` and `user_controller.py`).

Also, I think that parameter should also have a default of `0` like in the controller. We will probably reuse this function when we implement the "admin can search for users and give them special badges" feature.